design/ceph/object/ceph-sse-s3.md

```diff
@@ -0,0 +1,42 @@
+---
+# AWS server side encryption SSE-S3 support for RGW
+target-version: release-1.10
+---
+
+# Feature Name
+AWS server side encryption SSE-S3 support for RGW
+
+## Summary
+The S3 protocol supports three different types of [server side encryption](https://docs.aws.amazon.com/AmazonS3/latest/userguide/serv-side-encryption.html): SSE-C, SSE-KMS and SSE-S3. For the last two RGW server need to configure with external services such as [vault](https://www.vaultproject.io/). Currently Rook configure RGW with `SSE-KMS` options from v1.6 which needs atleast  cephv15, so that S3 request with `sse:kms` header will be handled properly. Recently the support for handling the `sse:s3` added to RGW(planning to support from cephv17 onwards) with another set of options, so Rook needs to provide option to configure RGW similar to `SSE-KMS`.
+
+### Goals
+Configure RGW with `SSE-S3` options, so that RGW can handle request with `sse:s3` headers.
+
+## Proposal details
+Introducing new field `s3` in `SecuritySpec` which defines `AWS-SSE-S3` support for RGW.
+```
+security:
+  kms:
+    ..
+  s3:
```

Comment:
According to [this comment](https://github.com/rook/rook/pull/10318#discussion_r882887830), I thought the `sse:s3` would not be enabled by the `security` section, but looks like now that is the proposal? It does seem good this way so we don't need a separate bool for enabling it.

Reply:
Earlier decided it was to include bool/enum in `ObjectStoreSpec` to decide the flavour RGW support and use settings from SecuritySpec. But later after the discussions, it felt right to include it in the `Security` session